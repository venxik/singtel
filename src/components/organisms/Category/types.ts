import { TouchableOpacityProps } from '../../atoms/Button';

export type CategoryProps = {
  text: string;
  isSelected: boolean;
} & TouchableOpacityProps;
