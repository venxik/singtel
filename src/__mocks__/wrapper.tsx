import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
