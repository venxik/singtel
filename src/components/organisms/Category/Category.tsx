import { StyleSheet, TouchableOpacity } from 'react-native';
import { ComicText, useThemeColor } from '../../atoms';
import { CategoryProps } from './types';

export const Category = (props: CategoryProps) => {
  const { style, lightColor, darkColor, isSelected, text, ...otherProps } = props;

  const backgroundColorSelected = useThemeColor(
    { light: lightColor, dark: darkColor },
    'categorySelected'
  );
  const backgroundColorNotSelected = useThemeColor(
    { light: lightColor, dark: darkColor },
    'categoryNotSelected'
  );
  const borderColor = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return (
    <TouchableOpacity
      style={[
        {
          backgroundColor: isSelected ? backgroundColorSelected : backgroundColorNotSelected,
          borderRadius: 20,
          paddingVertical: '5%',
          paddingHorizontal: '15%',
          borderWidth: 1,
          borderColor,
        },
        style,
      ]}
      {...otherProps}>
      <ComicText testID="category-text" style={styles.text}>
        {text}
      </ComicText>
    </TouchableOpacity>
  );
};

export default Category;

const styles = StyleSheet.create({
  text: { textAlign: 'center' },
});
