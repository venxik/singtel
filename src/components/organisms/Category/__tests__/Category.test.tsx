import React from 'react';
import { renderWithProviders } from '../../../../__mocks__';
import { fireEvent } from '@testing-library/react-native';
import Category from '../Category';

describe('Category', () => {
  it(`match snapshot`, () => {
    const tree = renderWithProviders(<Category text="test" isSelected />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`can be pressed and functional`, () => {
    let number = 1;
    const { getByTestId } = renderWithProviders(
      <Category testID="category-item" text="test" onPress={() => (number += 1)} isSelected />
    );
    fireEvent.press(getByTestId('category-item'));
    expect(number).toBe(2);
  });

  it(`can have correct text`, () => {
    const { getByTestId } = renderWithProviders(
      <Category testID="category-item" text="test" isSelected />
    );
    expect(getByTestId('category-text').props.children).toEqual('test');
  });

  it(`can have correct styling`, () => {
    const { getByTestId } = renderWithProviders(
      <Category testID="answer-item" text="test" isSelected />
    );
    expect(getByTestId('category-text').props.children).toEqual('test');
  });
});
