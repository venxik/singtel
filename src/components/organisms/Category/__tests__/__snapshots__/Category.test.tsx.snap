// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#686fcc",
      "borderColor": "#000",
      "borderRadius": 20,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": "15%",
      "paddingVertical": "5%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "textAlign": "center",
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="category-text"
  >
    test
  </Text>
</View>
`;

exports[`StyledText match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#686fcc",
      "borderColor": "#000",
      "borderRadius": 20,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": "15%",
      "paddingVertical": "5%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "textAlign": "center",
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="category-text"
  >
    test
  </Text>
</View>
`;
